experimental/generation/generator/src/library/dialogGenerator.ts

```diff
@@ -257,21 +259,23 @@ async function processTemplate(
                 let lgTemplate: lg.Templates | undefined = foundTemplate instanceof lg.Templates ? foundTemplate as lg.Templates : undefined
                 let plainTemplate: Plain | undefined = !lgTemplate ? foundTemplate as Plain : undefined
                 // Ignore templates that are defined, but are empty
-                if (plainTemplate?.source || lgTemplate?.allTemplates.some(f => f.name === 'template')) {
+                if (plainTemplate ?.source || lgTemplate ?.allTemplates.some(f => f.name === 'template')) {
                     // Constant file or .lg template so output
-                    feedback(FeedbackType.debug, `Using template ${plainTemplate ? plainTemplate.source : lgTemplate?.id}`)
+                    feedback(FeedbackType.debug, `Using template ${plainTemplate ? plainTemplate.source : lgTemplate ?.id}`)
 
                     let filename = addPrefix(scope.prefix, templateName)
-                    if (lgTemplate?.allTemplates.some(f => f.name === 'filename')) {
+                    if (lgTemplate ?.allTemplates.some(f => f.name === 'filename')) {
                         try {
                             filename = lgTemplate.evaluate('filename', scope) as string
                         } catch (e) {
                             throw new Error(`${templateName}: ${e.message}`)
                         }
-                    } else if (filename.includes(scope.locale)) {
+                    } else if (filename.includes(scope.locale) && !filename.includes('qna')) {
                         // Move constant files into locale specific directories
                         let prop = templateName.startsWith('library') ? 'library' : scope.property
                         filename = `${scope.locale}/${prop}/${filename}`
+                    } else if (filename.includes(scope.locale)) {
+                        filename = `${scope.locale}/QnA/${filename}`
```

Comment:
This should still be removed and the !filename.includes('qna') from the previous clause.  If you want the prop folder to be QnA put the test inside setting prop.